### PR TITLE
Add MachineConfigPool Metrics

### PR DESCRIPTION
**Pull request by troy0820**
- [x] MachineConfigPoolMetrics

### CODE REVIEW

File: pkg/monitor/cluster/cluster.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 
 	"github.com/Azure/go-autorest/autorest/azure"
 	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	mcpclient "github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned"
```
Comment (jim-minter) [inline pkg/monitor/cluster/cluster.go:14]: `mcoclient`

File: pkg/monitor/cluster/cluster.go [hunk 2]
```diff
@@ -28,6 +29,7 @@ type Monitor struct {
 	dims map[string]string
 
 	cli       kubernetes.Interface
+	mcp       mcpclient.Interface
```
Comment (jim-minter) [inline pkg/monitor/cluster/cluster.go:32]: mcocli

File: pkg/monitor/cluster/machineconfigpool.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import (
+	"context"
+
+	v1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var machineConfigPoolConditions = map[v1.MachineConfigPoolConditionType]struct{}{
```
Comment (jim-minter) [inline pkg/monitor/cluster/machineconfigpool.go:14]: machineConfigPoolConditionsWhitelist

File: pkg/monitor/cluster/machineconfigpool.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	v1.MachineConfigPoolUpdated:        {},
+	v1.MachineConfigPoolUpdating:       {},
+	v1.MachineConfigPoolNodeDegraded:   {},
+	v1.MachineConfigPoolRenderDegraded: {},
+	v1.MachineConfigPoolDegraded:       {},
+}
+
```
Comment (jim-minter) [inline pkg/monitor/cluster/machineconfigpool.go:21]: add machineConfigPoolNotConditions, should just include MachineConfigPoolUpdated

File: pkg/monitor/cluster/machineconfigpool.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+func (mon *Monitor) emitMachineConfigPoolMetrics(ctx context.Context) error {
+	mcps, err := mon.mcp.MachineconfigurationV1().MachineConfigPools().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	mon.emitGauge("machineconfigpools.count", int64(len(mcps.Items)), nil)
+
+	for _, machineConfigPools := range mcps.Items {
+		for _, mcp := range machineConfigPools.Status.Conditions {
```
Comment (jim-minter) [inline pkg/monitor/cluster/machineconfigpool.go:31]: here, check if you're in the whitelist.  If not, skip

File: pkg/monitor/cluster/machineconfigpool.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+			if _, ok := machineConfigPoolConditions[mcp.Type]; ok {
```
Comment (jim-minter) [inline pkg/monitor/cluster/machineconfigpool.go:32]: check if you're in machineConfigPoolNotConditions

File: pkg/monitor/cluster/machineconfigpool.go [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	v1.MachineConfigPoolDegraded:       {},
+}
+
+func (mon *Monitor) emitMachineConfigPoolMetrics(ctx context.Context) error {
+	mcps, err := mon.mcp.MachineconfigurationV1().MachineConfigPools().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	mon.emitGauge("machineconfigpools.count", int64(len(mcps.Items)), nil)
```
Comment (jim-minter) [inline pkg/monitor/cluster/machineconfigpool.go:28]: let's skip this for now

File: pkg/monitor/cluster/machineconfigpool_test.go [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+	m.EXPECT().EmitGauge("machineconfigpools.condition.count", int64(1), map[string]string{
+		"machineconfigpool": "machine-config-pool2",
+		"condition": "Updating",
+	})
+
+	err := mon.emitMachineConfigPoolMetrics(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
```
Comment (jim-minter) [inline pkg/monitor/cluster/machineconfigpool_test.go:72]: please set your editor to append a newline on each file!

Comment (troy0820) [reply pkg/monitor/cluster/machineconfigpool_test.go]: .editorconfig wasn't present to fix change on save 🤦

**Pull request merged**